task-driver/integration/helpers.rs

```diff
@@ -9,13 +9,66 @@ use common::types::{
     wallet_mocks::mock_empty_wallet,
 };
 use external_api::types::Wallet as ApiWallet;
-use eyre::Result;
+use eyre::{eyre, Result};
 use mpc_stark::algebra::scalar::Scalar;
 use rand::thread_rng;
 use renegade_crypto::hash::{evaluate_hash_chain, PoseidonCSPRNG};
 use starknet_client::client::StarknetClient;
 use system_bus::SystemBus;
-use task_driver::driver::{TaskDriver, TaskDriverConfig};
+use task_driver::{
+    driver::{TaskDriver, TaskDriverConfig},
+    lookup_wallet::LookupWalletTask,
+};
+use test_helpers::{assert_eq_result, assert_true_result};
+use uuid::Uuid;
+
+use crate::IntegrationTestArgs;
+
+// ---------
+// | Tasks |
+// ---------
+
+/// Lookup a wallet in the contract state and verify that it matches the expected wallet
+pub(crate) async fn lookup_wallet_and_check_result(
+    expected_wallet: &Wallet,
+    blinder_seed: Scalar,
+    share_seed: Scalar,
+    test_args: IntegrationTestArgs,
+) -> Result<()> {
+    // Start a lookup task for the new wallet
+    let new_wallet_id = Uuid::new_v4();
```

Comment:
shouldn't we be using the `expected_wallet.id` to look it up?

Reply:
No, this ID doesn't matter and doing it this way makes sure that we don't overwrite any state elements on lookup -- which is desired if the caller allocated state elements.